torch/csrc/cuda/Graph.cpp

```diff
@@ -48,14 +49,22 @@ void THCPGraph_init(PyObject* module) {
                   "Unknown capture error mode. Expected `global`, `thread_local`, or `relaxed`, got ",
                   capture_error_mode);
             }
+            self.set_prev_only_lift_cpu_tensors(
+                torch::utils::only_lift_cpu_tensors());
+            torch::utils::set_only_lift_cpu_tensors(true);
             return self.capture_begin(pool, capture_mode);
           },
           py::arg("pool"),
           py::arg("capture_error_mode"),
           py::call_guard<py::gil_scoped_release>())
       .def(
           "capture_end",
-          torch::wrap_pybind_function_no_gil(&at::cuda::CUDAGraph::capture_end))
+          [](::at::cuda::CUDAGraph& self) {
+            self.capture_end();
+            torch::utils::set_only_lift_cpu_tensors(
```

Comment:
This unfortunately is not sufficient as far as I know, since stream capture can span threads. And unfortunately autograd by default will use a separate thread from the main thread. I believe there is a way for the autograd thread to inherit the TorchDispatchModes of the main thread, which presumably can be applied here as well. But I'm not sure how that works internally.